src/SprykerEco/Zed/Econda/Business/EcondaBusinessFactory.php

```diff
@@ -0,0 +1,285 @@
+     * @return \SprykerEco\Zed\Econda\Business\Exporter\Writer\File\NameGenerator\CsvNameGenerator
+     */
+    protected function createCsvNameGenerator()
+    {
+        return new CsvNameGenerator();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Collector\File\EcondaCategoryCollector
+     */
+    public function createFileCategoryCollector()
+    {
+        $storageCategoryNodeCollector = new EcondaCategoryCollector(
+            $this->createCriteriaBuilder(),
+            $this->createStoragePdoQueryAdapterByName('CategoryNodeEcondaQuery')
+        );
+
+        return $storageCategoryNodeCollector;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Collector\File\EcondaProductCollector
+     */
+    public function createFileProductCollector()
+    {
+        /**
+         * @var \SprykerEco\Zed\Econda\Business\Collector\File\EcondaProductCollector $storageProductCollector
+         */
+        $storageProductCollector = new EcondaProductCollector(
+            $this->createCriteriaBuilder(),
+            $this->createStoragePdoQueryAdapterByName('ProductConcreteEcondaQuery'),
+            $this->getProductCategoryQueryContainer(),
+            $this->getProductImageQueryContainer(),
+            $this->getPriceFacade(),
+            $this->getConfig()
+        );
+
+        return $storageProductCollector;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Exporter\ExporterInterface
+     */
+    protected function createFileExporter()
+    {
+        $fileExporter = new FileExporter(
+            $this->createFileWriter(),
+            $this->createFailedResultModel(),
+            $this->createBatchResultModel(),
+            $this->createCsvNameGenerator(),
+            $this->getConfig()->getFileExportPath(),
+            $this->getCollectorFileExporterPlugins()
+        );
+
+        return $fileExporter;
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Dependency\Plugin\EcondaPluginInterface[]
+     */
+    protected function getCollectorFileExporterPlugins()
+    {
+        return $this->getProvidedDependency(EcondaDependencyProvider::FILE_PLUGINS);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Dependency\Facade\EcondaToLocaleInterface
+     */
+    protected function getLocaleFacade()
+    {
+        return $this->getProvidedDependency(EcondaDependencyProvider::FACADE_LOCALE);
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Persistence\EcondaQueryContainer
+     */
+    protected function getEcondaQueryContainer()
+    {
+        return new EcondaQueryContainer();
+    }
+
+    /**
+     * @return \SprykerEco\Zed\Econda\Business\Exporter\Writer\File\Adapter\CsvAdapter
+     */
+    protected function createCsvFileWriterAdapter()
+    {
+        return new CsvAdapter($this->getConfig()->getFileExportPath());
+    }
+
+    /**
+     * @return \Generated\Shared\Transfer\FailedResultTransfer
+     */
+    protected function createFailedResultModel()
```

Comment:
Model vs Transfer?
Transfers never have to be factorized, they can directly be instantiated inside code as those are generated classes.